src/view/com/util/post-embeds/VideoEmbedInner/VideoEmbedInnerWeb.tsx

```diff
@@ -77,12 +85,54 @@ export function VideoEmbedInnerWeb({
     }
   }, [embed.playlist])
 
+  // purge low quality segments from buffer on next frag change
+  useEffect(() => {
+    if (!hlsRef.current) return
+
+    if (focused) {
+      function fragChanged(
+        _event: Events.FRAG_CHANGED,
+        {frag}: FragChangedData,
+      ) {
+        if (!hlsRef.current) return
+
+        // if the current quality level goes above 0, flush the low quality segments
+        if (hlsRef.current.nextAutoLevel > 0) {
+          const flushed: Fragment[] = []
+
+          for (const lowQualFrag of lowQualityFragments) {
+            // avoid if close to the current fragment
+            if (Math.abs(frag.start - lowQualFrag.start) < 0.1) {
+              return
+            }
+
+            hlsRef.current.trigger(Hls.Events.BUFFER_FLUSHING, {
+              startOffset: lowQualFrag.start,
+              endOffset: lowQualFrag.end,
+              type: 'video',
+            })
+
+            flushed.push(lowQualFrag)
+          }
+
+          setLowQualityFragments(prev => prev.filter(f => !flushed.includes(f)))
+        }
+      }
+      hlsRef.current.on(Hls.Events.FRAG_CHANGED, fragChanged)
+
+      const current = hlsRef.current
```

Comment:
it would be best to capture this once at the top and then always use this reference instead of dereferencing during events. fewer things that can go wrong.

Reply:
Should I also use the captured version within the listener?